httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/gss/GssSchemeBase.java

```diff
@@ -183,35 +184,54 @@ public void processChallenge(
                 state = State.TOKEN_READY;
                 break;
             case TOKEN_SENT:
-                if (challenged) {
-                    state = State.TOKEN_READY;
-                } else if (mutualAuth) {
-                    // We should have received a valid mutualAuth token
-                    if (!gssContext.isEstablished()) {
+                if (challengeToken == null) {
+                    if (!challenged && ignoreMissingToken) {
+                        // Got a 200 without a challenge. Old non RFC compliant server.
```

Comment:
Why 200? The auth loop can be completed on any response status unless the intial token was incorrect.

Reply:
Fixing